lib/rinject.rb

```diff
@@ -0,0 +1,25 @@
+class Array
+
+  def retesh_inject(starter = nil)
+    array = self.clone
+    starter = array.shift if starter == nil
+    array.each do |element|
+      starter = yield(starter, element)
+    end
+    starter
+  end
+
+##Recursion - works but not tested
```

Comment:
Incorrect indentation detected (column 0 instead of 2).<br>Missing space after #.